packages/replay-next/src/suspense/ResumeTargetCache.ts

```diff
@@ -0,0 +1,110 @@
+import { ExecutionPoint, PauseDescription, SourceId, SourceLocation } from "@replayio/protocol";
+import { createCache } from "suspense";
+
+import { locationsInclude } from "protocol/utils";
+import { ReplayClientInterface } from "shared/client/types";
+
+import { PauseAndFrameId } from "../contexts/SelectedFrameContext";
+import { framesCache } from "./FrameCache";
+import { getPointAndTimeForPauseId, updateMappedLocation } from "./PauseCache";
+import { pointStackCache } from "./PointStackCache";
+import { sourceOutlineCache } from "./SourceOutlineCache";
+import { sourcesByIdCache } from "./SourcesCache";
+
+export const FIND_RESUME_TARGET_COMMANDS = ["findRewindTarget", "findResumeTarget"] as const;
+export const FIND_STEP_TARGET_COMMANDS = [
+  "findStepInTarget",
+  "findStepOutTarget",
+  "findStepOverTarget",
+  "findReverseStepOverTarget",
+] as const;
+export const FIND_TARGET_COMMANDS = [
+  ...FIND_RESUME_TARGET_COMMANDS,
+  ...FIND_STEP_TARGET_COMMANDS,
+] as const;
+export type FindTargetCommand = (typeof FIND_TARGET_COMMANDS)[number];
+
+const resumeTargetForPointCache = createCache<
+  [replayClient: ReplayClientInterface, command: FindTargetCommand, point: ExecutionPoint],
+  PauseDescription
+>({
+  debugLabel: "ResumeTargetForPoint",
+  getKey: ([, point, command]) => `${point}:${command}`,
+  load: async ([replayClient, command, point]) => {
+    const sources = await sourcesByIdCache.readAsync(replayClient);
+    const target = await replayClient[command](point);
+    if (target.frame) {
+      updateMappedLocation(sources, target.frame);
+    }
+    return target;
+  },
+});
+
+export const resumeTargetCache = createCache<
+  [
+    replayClient: ReplayClientInterface,
+    command: FindTargetCommand,
+    point: ExecutionPoint | null,
+    pauseAndFrameId: PauseAndFrameId | null,
+    sourceId: SourceId | undefined
+  ],
+  PauseDescription | undefined
+>({
+  debugLabel: "ResumeTarget",
+  getKey: ([, command, point, pauseAndFrameId, sourceId]) =>
+    `${command}:${point}:${pauseAndFrameId?.pauseId}:${pauseAndFrameId?.frameId}:${sourceId}`,
+  load: async ([replayClient, command, point, pauseAndFrameId, sourceId]) => {
+    if (!point) {
+      return;
+    }
+
+    let currentPoint = point;
+    if (pauseAndFrameId) {
+      const [pausePoint] = getPointAndTimeForPauseId(pauseAndFrameId.pauseId);
+      if (!pausePoint) {
+        return;
+      }
+
+      const frames = await framesCache.readAsync(replayClient, pauseAndFrameId.pauseId);
```

Comment:
I think we should return early here if `pauseAndFrameId.pauseId` is undefined (and not change the param or return types in the frames cache)